File: src/main/java/com/ccjiuhong/gui/swing/ConfigWindow.java
package com.ccjiuhong.gui.swing;

import lombok.extern.slf4j.Slf4j;

import javax.swing.*;
import java.awt.*;


/**
 * @author G. Seinfeld
 * @since 2019/07/01
 */
@Slf4j
class ConfigWindow extends JFrame {
    ConfigWindow() {
        setTitle("配置中心");

        setLocationRelativeTo(null);

        setSize(600, 480);

        setDefaultCloseOperation(EXIT_ON_CLOSE);

        JPanel jPanel = new JPanel();
        add(jPanel);

        JLabel downloadDirLabel = new JLabel("下载地址：");
        downloadDirLabel.setFont(new Font("Microsoft YaHei", Font.PLAIN, 16));
        jPanel.add(downloadDirLabel);

        JTextField downloadDirInput = new JTextField(20);
        downloadDirInput.setFont(new Font("Microsoft YaHei", Font.PLAIN, 16));
        jPanel.add(downloadDirInput);

        JButton downloadDirButton = new JButton("选择");
        downloadDirButton.setFont(new Font("Microsoft YaHei", Font.PLAIN, 16));
        downloadDirButton.addActionListener(e -> {
            JFrame frame = new JFrame();
            JFileChooser downloadDir = new JFileChooser();
            downloadDir.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
            frame.getContentPane().add(downloadDir);
            frame.setSize(600, 480);
            frame.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
            frame.setVisible(true);
            frame.requestFocus();
        });
        jPanel.add(downloadDirButton);

        JButton submit = new JButton("保存更改");
        jPanel.add(submit);

        setVisible(true);
    }

    public static void main(String[] args) {
        try {
            UIManager.setLookAndFeel("com.sun.java.swing.plaf.windows.WindowsLookAndFeel");
            new ConfigWindow();
        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException | UnsupportedLookAndFeelException e) {
            log.error(e.getMessage(), e);
        }
    }
}

File: src/main/java/com/ccjiuhong/gui/swing/ConfigWindow.kt
package com.ccjiuhong.gui.swing

import java.awt.Font
import java.awt.HeadlessException
import javax.swing.*

/**
 * @author G. Seinfeld
 * @date 2019/07/01
 */
class ConfigWindow @Throws(HeadlessException::class)
constructor() : JFrame() {
    init {
        title = "配置中心"

        setLocationRelativeTo(null)

        setSize(600, 480)

        defaultCloseOperation = EXIT_ON_CLOSE

        val jPanel = JPanel()
        add(jPanel)

        val downloadDirLabel = JLabel("下载地址：")
        downloadDirLabel.font = Font("Microsoft YaHei", Font.PLAIN, 16)
        jPanel.add(downloadDirLabel)


        val downloadDirInput = JTextField(20)
        downloadDirInput.font = Font("Microsoft YaHei", Font.PLAIN, 16)
        jPanel.add(downloadDirInput)

        val downloadDirButton = JButton("选择")
        downloadDirButton.font = Font("Microsoft YaHei", Font.PLAIN, 16)
        downloadDirButton.addActionListener {
            val frame = JFrame()
            val downloadDir = JFileChooser()
            downloadDir.fileSelectionMode = JFileChooser.DIRECTORIES_ONLY
            frame.contentPane.add(downloadDir)
            frame.setSize(600, 480)
            frame.defaultCloseOperation = WindowConstants.DISPOSE_ON_CLOSE
            frame.isVisible = true
            frame.requestFocus()
        }
        jPanel.add(downloadDirButton)

        val submit = JButton("保存更改")
        jPanel.add(submit)

        isVisible = true
    }
}

fun main() {
    try {
        UIManager.setLookAndFeel("com.sun.java.swing.plaf.windows.WindowsLookAndFeel")
    } catch (e: ClassNotFoundException) {
        logger.error(e.message, e)
    } catch (e: InstantiationException) {
        logger.error(e.message, e)
    } catch (e: IllegalAccessException) {
        logger.error(e.message, e)
    } catch (e: UnsupportedLookAndFeelException) {
        logger.error(e.message, e)
    }
    ConfigWindow()
}